The 0.16.0 version fails with "Could not initialize class com.google.cloud.spark.bigquery.SparkBigQueryConnectorUserAgentProvider"

The current latest version fails to fetch data  from BigQuery with 
`java.lang.NoClassDefFoundError: Could not initialize class com.google.cloud.spark.bigquery.SparkBigQueryConnectorUserAgentProvider`

It can get basic metadata, so the data frames get created, but once we need to materialize or compute it fails.

The exact fat jar used: https://storage.googleapis.com/spark-lib/bigquery/spark-bigquery-latest_2.11.jar and it passed to the spark-submit with the `--jars`

The code is a simple spark load: 
`spark.read.format("bigquery").option("table", "project.dataset.tablename").load()`

The full stack trace relevant to the connector code: 
```
Caused by: java.lang.NoClassDefFoundError: Could not initialize class com.google.cloud.spark.bigquery.SparkBigQueryConnectorUserAgentProvider
	at com.google.cloud.spark.bigquery.direct.DirectBigQueryRelation$.headerProvider(DirectBigQueryRelation.scala:356)
	at com.google.cloud.spark.bigquery.direct.DirectBigQueryRelation$.createReadClient(DirectBigQueryRelation.scala:333)
	at com.google.cloud.spark.bigquery.direct.DirectBigQueryRelation$$anonfun$$lessinit$greater$default$3$1.apply(DirectBigQueryRelation.scala:42)
	at com.google.cloud.spark.bigquery.direct.DirectBigQueryRelation$$anonfun$$lessinit$greater$default$3$1.apply(DirectBigQueryRelation.scala:42)
	at com.google.cloud.spark.bigquery.direct.BigQueryRDD.compute(BigQueryRDD.scala:46)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at .....
```

--- a/connector/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConnectorUserAgentProvider.java
+++ b/connector/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConnectorUserAgentProvider.java
@@ -24,7 +24,6 @@ import org.apache.http.client.methods.CloseableHttpResponse;
 import org.apache.http.client.methods.HttpGet;
 import org.apache.http.impl.client.CloseableHttpClient;
 import org.apache.http.impl.client.HttpClients;
-import org.apache.spark.sql.SparkSession;
 import scala.util.Properties;
 
 import java.io.IOException;
@@ -46,7 +45,8 @@ public class SparkBigQueryConnectorUserAgentProvider implements UserAgentProvide
             .map(image -> " dataproc-image/" + image)
             .orElse("");
     private static String CONNECTOR_VERSION = BuildInfo.version();
-    private static String SPARK_VERSION = SparkSession.builder().getOrCreate().version();
+    // In order to avoid using SparkContext or SparkSession, we are going directly to the source
+    private static String SPARK_VERSION = org.apache.spark.package$.MODULE$.SPARK_VERSION();
     private static String JAVA_VERSION = System.getProperty("java.runtime.version");
     private static String SCALA_VERSION = Properties.versionNumberString();
     static final String USER_AGENT = format("spark-bigquery-connector/%s spark/%s java/%s scala/%s%s%s",

